api/agent.go

```diff
@@ -322,15 +333,60 @@ func NewControllerWrapper(config *loadtest.Config, controllerConfig interface{},
 		}
 	}
 
+	var creds []string
+	f, err := os.Open(config.UsersConfiguration.UsersFilePath)
+	if err == nil {
+		// The file is an optional parameter which may not be present.
+		// So we read it only if it opens successfully.
+		defer f.Close()
+		scanner := bufio.NewScanner(f)
+		for scanner.Scan() {
+			creds = append(creds, scanner.Text())
+		}
+		if err := scanner.Err(); err != nil {
+			return nil, fmt.Errorf("failed to read from %s: %w", f.Name(), err)
+		}
+	} else if !os.IsNotExist(err) {
+		// If the file exists, but failed to open, then probably it is worthwhile
+		// logging it.
+		mlog.Warn("Failed to open UsersFilePath. Continuing with default signup", mlog.Err(err))
+	}
+
 	return func(id int, status chan<- control.UserStatus) (control.UserController, error) {
 		id += userOffset
 
+		var username, password, email string
+
+		if len(creds) > 0 {
+			// We do not allow to mix user-defined and auto signup modes.
+			// If the user specifies a custom login file, then it should contain
+			// all the users that is expected to login during a load-test.
+			if id >= len(creds) {
+				return nil, errors.New("exceeded user list limit")
+			}
```

Comment:
I think we should prevent failing in the middle of a load-test when possible. We have access to the config so we know how many users we can potentially add (`MaxActiveUsers`) and if that value doesn't match our expectations we should probably fail before starting anything.